### PR TITLE
Docs/component interaction

### PR DESCRIPTION
**Pull request by thomas-mauran**

### CODE REVIEW

**Review by BioTheWolff — DISMISSED** [review]
LGTM!

File: Architecture.md [hunk 1]
```diff
@@ -48,35 +47,103 @@
...
+
+        pom -. manages deployments .-> docker
+
+        docker -. use images .-> cr
+
+        cr === fs
+    end
+```
+
+In this schema we can see many different component interacting with each other. We decided for this first iteration to split services as much as we could to allow for a more modular architecture. This allows us to easily replace a component if needed, and to scale the different components independently.
```
Comment (Kuruyia) [inline Architecture.md:91]:
```suggestion
In this schema, we can see the different components interacting with each other. We decided, for this first iteration, to split services as much as we could to allow for a more modular architecture. This allows us to easily replace a component if needed, and to scale the different components independently.
```

File: Architecture.md [hunk 2]
```diff
@@ -48,35 +47,103 @@
...
+
+- The client API is the main component that will be used by the clients to interact with the service. This API will then interact with the git controller to create repositories and with pomegranate to create docker deployments, and with the database to store the data.
```
Comment (Kuruyia) [inline Architecture.md:93]:
```suggestion
- The client API is the main component that will be used by the clients to interact with the service. This API will then interact with the Git controller to create repositories and with Pomegranate to create Docker deployments, and with the database to store the data.
```

File: Architecture.md [hunk 3]
```diff
@@ -48,35 +47,103 @@
...
+- The git controller will interact with the host filesystem to create repositories, and with the local container registry to push images.
```
Comment (Kuruyia) [inline Architecture.md:94]:
```suggestion
- The Git controller will interact with the host file system to create repositories, and with the local container registry to push images.
```

File: Architecture.md [hunk 4]
```diff
@@ -48,35 +47,103 @@
...
+- Pomegranate, our deployment manager, will interact with the local container registry to pull images and with the host filesystem to create deployments.
```
Comment (Kuruyia) [inline Architecture.md:95]:
```suggestion
- Pomegranate, our deployment manager, will interact with the local container registry to pull images and with the host file system to create deployments.
```

File: Architecture.md [hunk 5]
```diff
@@ -48,35 +47,103 @@
...
+
+Most of the interactions between the different components are done through gRPC, except for the web frontend and the CLI, which use HTTP requests. Using gRPC allows for a more efficient communication between the components, as well as a more secure one, as the communication is done through a private network. While the http requests allows us easier access to the API.
```
Comment (Kuruyia) [inline Architecture.md:97]:
```suggestion
Most of the interactions between the different components are done through gRPC, except for the web frontend and the CLI, which use HTTP requests. Using gRPC allows for a more efficient communication between the components, as well as a more secure one, as the communication is done through a private network. While the HTTP requests allows us easier access to the API.
```

**Review by Kuruyia — CHANGES_REQUESTED** [review]

**Review by Mathias-Boulay — CHANGES_REQUESTED** [review]
The idea is good, however a few suggestions:
- most sentences need to be straightforward. Go straight to the point.
- Using gRPC doesn't mean our communications are intrinsically secured.

**Comment by Mathias-Boulay** [comment]
Also, I forgot to mention but the graph background colors aren't the best fit for the official gh dark theme. Perhaps use a desaturated palette ?

**Review by Mathias-Boulay — APPROVED** [review]
Nice :+1:

**Review by BioTheWolff — APPROVED** [review]
LGTM!

**Review by Mathias-Boulay — APPROVED** [review]
LGTM :+1:

**Pull request merged**